src/components/GraphiQL.js

```diff
@@ -895,6 +897,25 @@ function isPromise(value) {
   return typeof value === 'object' && typeof value.then === 'function';
 }
 
+function observableToPromise(observable) {
+  if ( isPromise(observable) ) {
+    return observable;
+  }
+  if ( !isObservable(observable) ) {
+    return observable;
+  }
+  return new Promise((resolve, reject) => {
```

Comment:
Ok, Gotcha, so in that case unsubscribe should be called as well.
also, reject won't give any helpful information.
I've added a commit with both CR notes. :)